server/src/main/kotlin/com/trib3/server/graphql/GraphQLWebSocketProtocol.kt

```diff
@@ -23,6 +45,7 @@ data class OperationMessage<T : Any>(
  */
 data class OperationType<T : Any>(
     @get:JsonValue val type: String,
+    @JsonIgnore
```

Comment:
whats this?

Reply:
those are jackson annotations that control how the class is serialized/deserialized into/out of JSON.  specifically for `OperationType`:

`@JsonValue`:  serialize an `OperationType` as if it's just its string `type`
`@JsonIgnore`: don't serialize the `payloadType` member (might not be necessary with `JsonValue`, but not sure)
`@JsonCreator`: use this method as a factory method for deserializing an `OperationType`

so basically, for example `OperationType("start", GraphRequest::class)` serializes as the string `start`.  and, for example, deserializing the string `start` will call the static `getOperationType()` method to deserialize to an operation type that already has the appropriate `GraphRequest::class` `payloadType` set.

The polymorphic annotations on `OperationMessage` are a little more complicated, but basically tell it to look at the `type` field to determine the type of the `payload` and deserialize it appropriately in to the correctly typed objects.